{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-03-18T21:49:54.213699Z",
     "start_time": "2020-03-18T21:49:52.457031Z"
    }
   },
   "outputs": [],
   "source": [
    "import random\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from scipy.special import comb\n",
    "import scipy as sp\n",
    "from pylab import *\n",
    "import numpy as np\n",
    "from matplotlib import pyplot as plt\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import collections\n",
    "from IPython.display import clear_output\n",
    "import random\n",
    "sns.set_style(\"ticks\")\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-03-18T21:57:35.411643Z",
     "start_time": "2020-03-18T21:57:35.352582Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>specimen</th>\n",
       "      <th>menses</th>\n",
       "      <th>nugent</th>\n",
       "      <th>amsel</th>\n",
       "      <th>pH</th>\n",
       "      <th>vag_fluid</th>\n",
       "      <th>clue</th>\n",
       "      <th>whiff</th>\n",
       "      <th>race</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>p4z2r18</td>\n",
       "      <td>no</td>\n",
       "      <td>10</td>\n",
       "      <td>pos</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Abnormal</td>\n",
       "      <td>many</td>\n",
       "      <td>pos</td>\n",
       "      <td>White</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>p4z2r38</td>\n",
       "      <td>no</td>\n",
       "      <td>8</td>\n",
       "      <td>pos</td>\n",
       "      <td>5.3</td>\n",
       "      <td>Abnormal</td>\n",
       "      <td>many</td>\n",
       "      <td>neg</td>\n",
       "      <td>Black</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>p4z2r15</td>\n",
       "      <td>no</td>\n",
       "      <td>0</td>\n",
       "      <td>neg</td>\n",
       "      <td>4.4</td>\n",
       "      <td>Normal</td>\n",
       "      <td>absent</td>\n",
       "      <td>neg</td>\n",
       "      <td>Black</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>p4z2r39</td>\n",
       "      <td>no</td>\n",
       "      <td>0</td>\n",
       "      <td>neg</td>\n",
       "      <td>4.4</td>\n",
       "      <td>Normal</td>\n",
       "      <td>absent</td>\n",
       "      <td>neg</td>\n",
       "      <td>White</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>p4z1r18</td>\n",
       "      <td>no</td>\n",
       "      <td>0</td>\n",
       "      <td>neg</td>\n",
       "      <td>4.0</td>\n",
       "      <td>Abnormal</td>\n",
       "      <td>absent</td>\n",
       "      <td>neg</td>\n",
       "      <td>White</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>215</td>\n",
       "      <td>p4z2r26</td>\n",
       "      <td>no</td>\n",
       "      <td>8</td>\n",
       "      <td>pos</td>\n",
       "      <td>5.3</td>\n",
       "      <td>Abnormal</td>\n",
       "      <td>many</td>\n",
       "      <td>pos</td>\n",
       "      <td>White</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>216</td>\n",
       "      <td>p4z2r27</td>\n",
       "      <td>no</td>\n",
       "      <td>8</td>\n",
       "      <td>pos</td>\n",
       "      <td>5.6</td>\n",
       "      <td>Abnormal</td>\n",
       "      <td>few</td>\n",
       "      <td>pos</td>\n",
       "      <td>Black</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>217</td>\n",
       "      <td>p4z1r2</td>\n",
       "      <td>no</td>\n",
       "      <td>6</td>\n",
       "      <td>neg</td>\n",
       "      <td>5.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>absent</td>\n",
       "      <td>neg</td>\n",
       "      <td>White</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>218</td>\n",
       "      <td>p4z2r1</td>\n",
       "      <td>yes</td>\n",
       "      <td>8</td>\n",
       "      <td>pos</td>\n",
       "      <td>5.3</td>\n",
       "      <td>Abnormal</td>\n",
       "      <td>many</td>\n",
       "      <td>pos</td>\n",
       "      <td>Black</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>219</td>\n",
       "      <td>p4z2r30</td>\n",
       "      <td>no</td>\n",
       "      <td>9</td>\n",
       "      <td>pos</td>\n",
       "      <td>5.3</td>\n",
       "      <td>Abnormal</td>\n",
       "      <td>many</td>\n",
       "      <td>pos</td>\n",
       "      <td>Black</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>220 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    specimen menses  nugent amsel   pH vag_fluid    clue whiff   race\n",
       "0    p4z2r18     no      10   pos  NaN  Abnormal    many   pos  White\n",
       "1    p4z2r38     no       8   pos  5.3  Abnormal    many   neg  Black\n",
       "2    p4z2r15     no       0   neg  4.4    Normal  absent   neg  Black\n",
       "3    p4z2r39     no       0   neg  4.4    Normal  absent   neg  White\n",
       "4    p4z1r18     no       0   neg  4.0  Abnormal  absent   neg  White\n",
       "..       ...    ...     ...   ...  ...       ...     ...   ...    ...\n",
       "215  p4z2r26     no       8   pos  5.3  Abnormal    many   pos  White\n",
       "216  p4z2r27     no       8   pos  5.6  Abnormal     few   pos  Black\n",
       "217   p4z1r2     no       6   neg  5.6       NaN  absent   neg  White\n",
       "218   p4z2r1    yes       8   pos  5.3  Abnormal    many   pos  Black\n",
       "219  p4z2r30     no       9   pos  5.3  Abnormal    many   pos  Black\n",
       "\n",
       "[220 rows x 9 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_path = \"/Users/mzhu/Documents/Python/2020_20_440/440_project/\"\n",
    "Srini = pd.read_excel(file_path+\"pone.0037818.s006.xlsx\")\n",
    "pd.read_excel(file_path+\"pone.0037818.s007.xlsx\")\n",
    "pd.read_excel(file_path+\"pone.0037818.s005.xlsx\")\n",
    "pd.read_excel(file_path+\"st04.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-03-18T21:58:34.768874Z",
     "start_time": "2020-03-18T21:58:33.777035Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>p4z1r8</th>\n",
       "      <th>p1z1r2</th>\n",
       "      <th>p2z1r36</th>\n",
       "      <th>p2z1r40</th>\n",
       "      <th>p2z1r67</th>\n",
       "      <th>p3z1r14</th>\n",
       "      <th>p3z1r25</th>\n",
       "      <th>p3z1r28</th>\n",
       "      <th>p3z1r29</th>\n",
       "      <th>...</th>\n",
       "      <th>p3z2r65</th>\n",
       "      <th>p3z2r75</th>\n",
       "      <th>p3z2r8</th>\n",
       "      <th>p4z1r34</th>\n",
       "      <th>p4z1r6</th>\n",
       "      <th>p4z1r76</th>\n",
       "      <th>p4z2r10</th>\n",
       "      <th>p4z2r18</th>\n",
       "      <th>p4z2r37</th>\n",
       "      <th>Unnamed: 221</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>samplename</td>\n",
       "      <td>1130.0</td>\n",
       "      <td>1239</td>\n",
       "      <td>1126</td>\n",
       "      <td>1137</td>\n",
       "      <td>1128</td>\n",
       "      <td>1211</td>\n",
       "      <td>1326</td>\n",
       "      <td>1284</td>\n",
       "      <td>1290</td>\n",
       "      <td>...</td>\n",
       "      <td>1133</td>\n",
       "      <td>1184</td>\n",
       "      <td>1202</td>\n",
       "      <td>1171</td>\n",
       "      <td>1218</td>\n",
       "      <td>1187</td>\n",
       "      <td>1291</td>\n",
       "      <td>1090</td>\n",
       "      <td>1226</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>nugent</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>amsel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>neg</td>\n",
       "      <td>neg</td>\n",
       "      <td>neg</td>\n",
       "      <td>neg</td>\n",
       "      <td>neg</td>\n",
       "      <td>neg</td>\n",
       "      <td>neg</td>\n",
       "      <td>neg</td>\n",
       "      <td>...</td>\n",
       "      <td>pos</td>\n",
       "      <td>pos</td>\n",
       "      <td>pos</td>\n",
       "      <td>pos</td>\n",
       "      <td>pos</td>\n",
       "      <td>pos</td>\n",
       "      <td>pos</td>\n",
       "      <td>pos</td>\n",
       "      <td>pos</td>\n",
       "      <td>Total numbers of reads</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Lactobacillus iners</td>\n",
       "      <td>1207.0</td>\n",
       "      <td>2157</td>\n",
       "      <td>912</td>\n",
       "      <td>2587</td>\n",
       "      <td>90</td>\n",
       "      <td>748</td>\n",
       "      <td>1057</td>\n",
       "      <td>439</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>147</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>143</td>\n",
       "      <td>0</td>\n",
       "      <td>128829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Lactobacillus crispatus</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13330</td>\n",
       "      <td>113</td>\n",
       "      <td>56</td>\n",
       "      <td>461</td>\n",
       "      <td>293</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>900</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>79005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>154</td>\n",
       "      <td>Slackia exigua</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>155</td>\n",
       "      <td>Streptococcus</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>156</td>\n",
       "      <td>Streptococcus parasanguinis</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>157</td>\n",
       "      <td>Varibaculum cambriense</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>158</td>\n",
       "      <td>Number of reads per sample</td>\n",
       "      <td>1462.0</td>\n",
       "      <td>15551</td>\n",
       "      <td>1841</td>\n",
       "      <td>2735</td>\n",
       "      <td>579</td>\n",
       "      <td>1116</td>\n",
       "      <td>1058</td>\n",
       "      <td>562</td>\n",
       "      <td>904</td>\n",
       "      <td>...</td>\n",
       "      <td>954</td>\n",
       "      <td>1126</td>\n",
       "      <td>1120</td>\n",
       "      <td>1352</td>\n",
       "      <td>2080</td>\n",
       "      <td>1966</td>\n",
       "      <td>2420</td>\n",
       "      <td>3127</td>\n",
       "      <td>3239</td>\n",
       "      <td>425775</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>159 rows × 222 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Unnamed: 0  p4z1r8 p1z1r2 p2z1r36 p2z1r40 p2z1r67  \\\n",
       "0                     samplename  1130.0   1239    1126    1137    1128   \n",
       "1                         nugent     8.0      0       0       0       0   \n",
       "2                          amsel     NaN    neg     neg     neg     neg   \n",
       "3            Lactobacillus iners  1207.0   2157     912    2587      90   \n",
       "4        Lactobacillus crispatus     0.0  13330     113      56     461   \n",
       "..                           ...     ...    ...     ...     ...     ...   \n",
       "154               Slackia exigua     0.0      0       0       0       0   \n",
       "155                Streptococcus     0.0      0       0       0       0   \n",
       "156  Streptococcus parasanguinis     0.0      0       0       0       0   \n",
       "157       Varibaculum cambriense     0.0      0       0       0       0   \n",
       "158   Number of reads per sample  1462.0  15551    1841    2735     579   \n",
       "\n",
       "    p3z1r14 p3z1r25 p3z1r28 p3z1r29  ... p3z2r65 p3z2r75 p3z2r8 p4z1r34  \\\n",
       "0      1211    1326    1284    1290  ...    1133    1184   1202    1171   \n",
       "1         0       0       0       0  ...      10      10     10      10   \n",
       "2       neg     neg     neg     neg  ...     pos     pos    pos     pos   \n",
       "3       748    1057     439       2  ...      16       0     12       1   \n",
       "4       293       0       3     900  ...       0       0      0       1   \n",
       "..      ...     ...     ...     ...  ...     ...     ...    ...     ...   \n",
       "154       0       0       0       0  ...       0       0      0       0   \n",
       "155       0       0       0       0  ...       0       0      0       0   \n",
       "156       0       0       0       0  ...       0       0      0       0   \n",
       "157       0       0       0       0  ...       0       0      0       0   \n",
       "158    1116    1058     562     904  ...     954    1126   1120    1352   \n",
       "\n",
       "    p4z1r6 p4z1r76 p4z2r10 p4z2r18 p4z2r37            Unnamed: 221  \n",
       "0     1218    1187    1291    1090    1226                     NaN  \n",
       "1       10      10      10      10      10                     NaN  \n",
       "2      pos     pos     pos     pos     pos  Total numbers of reads  \n",
       "3      147       0       2     143       0                  128829  \n",
       "4        0       0       0       0       0                   79005  \n",
       "..     ...     ...     ...     ...     ...                     ...  \n",
       "154      0       0       0       0       0                       1  \n",
       "155      0       0       0       0       0                       1  \n",
       "156      0       0       0       0       0                       1  \n",
       "157      0       0       0       0       0                       1  \n",
       "158   2080    1966    2420    3127    3239                  425775  \n",
       "\n",
       "[159 rows x 222 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_excel(file_path+\"pone.0037818.s007.xlsx\")\n",
    "pd.read_excel(file_path+\"pone.0037818.s005.xlsx\")\n",
    "pd.read_excel(file_path+\"st04.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-03-18T21:51:36.508671Z",
     "start_time": "2020-03-18T21:51:36.470608Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tax_name</th>\n",
       "      <th>tax_id</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Acinetobacter baumannii</td>\n",
       "      <td>470</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Acinetobacter calcoaceticus</td>\n",
       "      <td>471</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Actinobaculum massiliense</td>\n",
       "      <td>202789</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Actinomyces hongkongensis</td>\n",
       "      <td>178339</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Actinomyces lingnae</td>\n",
       "      <td>114702</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>346</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>347</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>348</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>349</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>350</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>796.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>351 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        tax_name  tax_id  count\n",
       "0        Acinetobacter baumannii     470    5.0\n",
       "1    Acinetobacter calcoaceticus     471    5.0\n",
       "2      Actinobaculum massiliense  202789    5.0\n",
       "3      Actinomyces hongkongensis  178339    1.0\n",
       "4            Actinomyces lingnae  114702    1.0\n",
       "..                           ...     ...    ...\n",
       "346                          NaN     NaN    NaN\n",
       "347                          NaN     NaN    NaN\n",
       "348                          NaN     NaN    NaN\n",
       "349                          NaN     NaN    NaN\n",
       "350                          NaN     NaN  796.0\n",
       "\n",
       "[351 rows x 3 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_excel(file_path+\"pone.0037818.s005.xlsx\")\n",
    "pd.read_excel(file_path+\"st04.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-03-18T21:51:53.496267Z",
     "start_time": "2020-03-18T21:51:52.402067Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Table S4. Subject metadata and vaginal bacterial community composition.</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "      <th>Unnamed: 8</th>\n",
       "      <th>Unnamed: 9</th>\n",
       "      <th>...</th>\n",
       "      <th>Unnamed: 245</th>\n",
       "      <th>Unnamed: 246</th>\n",
       "      <th>Unnamed: 247</th>\n",
       "      <th>Unnamed: 248</th>\n",
       "      <th>Unnamed: 249</th>\n",
       "      <th>Unnamed: 250</th>\n",
       "      <th>Unnamed: 251</th>\n",
       "      <th>Unnamed: 252</th>\n",
       "      <th>Unnamed: 253</th>\n",
       "      <th>Unnamed: 254</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Percentage of 16S rRNA gene sequence reads ass...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Subject ID</td>\n",
       "      <td>Ethnic Groupa</td>\n",
       "      <td>pH</td>\n",
       "      <td>Nugent score</td>\n",
       "      <td>Nugent score categoryb</td>\n",
       "      <td>Community groupc</td>\n",
       "      <td>Total number readsd</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L. iners</td>\n",
       "      <td>L. crispatus</td>\n",
       "      <td>...</td>\n",
       "      <td>Patulibacter</td>\n",
       "      <td>Gordonia</td>\n",
       "      <td>Flectobacillus</td>\n",
       "      <td>Dermacoccus</td>\n",
       "      <td>Bacillus_c</td>\n",
       "      <td>Clostridiales_18</td>\n",
       "      <td>Proteobacteria_16</td>\n",
       "      <td>Simplicispira</td>\n",
       "      <td>Bacilli_2</td>\n",
       "      <td>Microbacterium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>S001</td>\n",
       "      <td>Asian</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>low</td>\n",
       "      <td>I</td>\n",
       "      <td>2335</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.44</td>\n",
       "      <td>90.84</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>S002</td>\n",
       "      <td>White</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>low</td>\n",
       "      <td>II</td>\n",
       "      <td>1403</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>S003</td>\n",
       "      <td>Black</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>low</td>\n",
       "      <td>III</td>\n",
       "      <td>1376</td>\n",
       "      <td>NaN</td>\n",
       "      <td>74.2</td>\n",
       "      <td>0.07</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>391</td>\n",
       "      <td>S390</td>\n",
       "      <td>Asian</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>intermediate</td>\n",
       "      <td>III</td>\n",
       "      <td>1866</td>\n",
       "      <td>NaN</td>\n",
       "      <td>72.19</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>392</td>\n",
       "      <td>S391</td>\n",
       "      <td>Asian</td>\n",
       "      <td>4.4</td>\n",
       "      <td>0</td>\n",
       "      <td>low</td>\n",
       "      <td>III</td>\n",
       "      <td>1731</td>\n",
       "      <td>NaN</td>\n",
       "      <td>76.31</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>393</td>\n",
       "      <td>S392</td>\n",
       "      <td>Asian</td>\n",
       "      <td>4.4</td>\n",
       "      <td>0</td>\n",
       "      <td>low</td>\n",
       "      <td>IV</td>\n",
       "      <td>1967</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>29.79</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>394</td>\n",
       "      <td>S393</td>\n",
       "      <td>Asian</td>\n",
       "      <td>4.4</td>\n",
       "      <td>0</td>\n",
       "      <td>low</td>\n",
       "      <td>III</td>\n",
       "      <td>1682</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.46</td>\n",
       "      <td>0.12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>395</td>\n",
       "      <td>S394</td>\n",
       "      <td>Asian</td>\n",
       "      <td>4.4</td>\n",
       "      <td>2</td>\n",
       "      <td>low</td>\n",
       "      <td>III</td>\n",
       "      <td>1725</td>\n",
       "      <td>NaN</td>\n",
       "      <td>93.39</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>396 rows × 255 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Table S4. Subject metadata and vaginal bacterial community composition.   \\\n",
       "0                                             Metadata                         \n",
       "1                                           Subject ID                         \n",
       "2                                                 S001                         \n",
       "3                                                 S002                         \n",
       "4                                                 S003                         \n",
       "..                                                 ...                         \n",
       "391                                               S390                         \n",
       "392                                               S391                         \n",
       "393                                               S392                         \n",
       "394                                               S393                         \n",
       "395                                               S394                         \n",
       "\n",
       "        Unnamed: 1 Unnamed: 2    Unnamed: 3              Unnamed: 4  \\\n",
       "0              NaN        NaN           NaN                     NaN   \n",
       "1    Ethnic Groupa         pH  Nugent score  Nugent score categoryb   \n",
       "2            Asian          4             0                     low   \n",
       "3            White          4             0                     low   \n",
       "4            Black          4             1                     low   \n",
       "..             ...        ...           ...                     ...   \n",
       "391          Asian          7             6            intermediate   \n",
       "392          Asian        4.4             0                     low   \n",
       "393          Asian        4.4             0                     low   \n",
       "394          Asian        4.4             0                     low   \n",
       "395          Asian        4.4             2                     low   \n",
       "\n",
       "            Unnamed: 5           Unnamed: 6  Unnamed: 7  \\\n",
       "0                  NaN                  NaN         NaN   \n",
       "1    Community groupc   Total number readsd         NaN   \n",
       "2                    I                 2335         NaN   \n",
       "3                   II                 1403         NaN   \n",
       "4                  III                 1376         NaN   \n",
       "..                 ...                  ...         ...   \n",
       "391                III                 1866         NaN   \n",
       "392                III                 1731         NaN   \n",
       "393                 IV                 1967         NaN   \n",
       "394                III                 1682         NaN   \n",
       "395                III                 1725         NaN   \n",
       "\n",
       "                                            Unnamed: 8    Unnamed: 9  ...  \\\n",
       "0    Percentage of 16S rRNA gene sequence reads ass...           NaN  ...   \n",
       "1                                             L. iners  L. crispatus  ...   \n",
       "2                                                 8.44         90.84  ...   \n",
       "3                                                 0.14             0  ...   \n",
       "4                                                 74.2          0.07  ...   \n",
       "..                                                 ...           ...  ...   \n",
       "391                                              72.19             0  ...   \n",
       "392                                              76.31             0  ...   \n",
       "393                                                  0         29.79  ...   \n",
       "394                                              99.46          0.12  ...   \n",
       "395                                              93.39             0  ...   \n",
       "\n",
       "     Unnamed: 245 Unnamed: 246    Unnamed: 247 Unnamed: 248 Unnamed: 249  \\\n",
       "0             NaN          NaN             NaN          NaN          NaN   \n",
       "1    Patulibacter     Gordonia  Flectobacillus  Dermacoccus   Bacillus_c   \n",
       "2               0            0               0            0            0   \n",
       "3               0            0               0            0            0   \n",
       "4               0            0               0            0            0   \n",
       "..            ...          ...             ...          ...          ...   \n",
       "391             0            0               0            0            0   \n",
       "392             0            0               0            0            0   \n",
       "393             0            0               0            0            0   \n",
       "394             0            0               0            0            0   \n",
       "395             0            0               0            0            0   \n",
       "\n",
       "         Unnamed: 250       Unnamed: 251   Unnamed: 252 Unnamed: 253  \\\n",
       "0                 NaN                NaN            NaN          NaN   \n",
       "1    Clostridiales_18  Proteobacteria_16  Simplicispira    Bacilli_2   \n",
       "2                   0                  0              0            0   \n",
       "3                   0                  0              0            0   \n",
       "4                   0                  0              0            0   \n",
       "..                ...                ...            ...          ...   \n",
       "391                 0                  0              0            0   \n",
       "392                 0                  0              0            0   \n",
       "393                 0                  0              0            0   \n",
       "394                 0                  0              0            0   \n",
       "395                 0                  0              0            0   \n",
       "\n",
       "       Unnamed: 254  \n",
       "0               NaN  \n",
       "1    Microbacterium  \n",
       "2                 0  \n",
       "3                 0  \n",
       "4                 0  \n",
       "..              ...  \n",
       "391               0  \n",
       "392               0  \n",
       "393               0  \n",
       "394               0  \n",
       "395               0  \n",
       "\n",
       "[396 rows x 255 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_excel(file_path+\"st04.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
